### PR TITLE
Fixed errors logging in dump.go

### PR DESCRIPTION
**Pull request by Rychu-Pawel**
This is regarding https://github.com/go-gitea/gitea/issues/9111
Changed the logger to the internal one that correctly closes the file stream before `os.Exit(1)`. Aditionally printing the errors also to the `stderr`.

### CODE REVIEW

**Comment by guillep2k** [comment]
Please run `make fmt`, commit and push the results.

**Comment by techknowlogick** [comment]
I think the Gitea log package has a fatalf function, if not could you add it there so that all packages using it can makes use of this additional function?

File: cmd/dump.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ It can be used for backup and capture Gitea server image to send to maintainer`,
 	},
 }
 
+func fatal(format string, args ...interface{}) {
```
Comment (lafriks) [inline cmd/dump.go:58]: is this really needed? Also why then in some places this function is used while in others not

Comment (Rychu-Pawel) [reply cmd/dump.go:58]: My intent was to give error feedback also to a console so users don't have dig through logs to find an issue. I guess this method is better than adding additional `fmt.Fprintf` everywhere before `log.Fatal`.
Also logging to stderr should help troubleshooting when `dump` is run from the cron.
What do you think?

**Comment by Rychu-Pawel** [comment]
@techknowlogick yes it has, this is the one I'm using. I preformatted the text to the variable there to be able to use `fmt.Fprintln` without concatination with the new line but I has just changed that so I guess it looks less confusing now.

File: cmd/dump.go (hunk 1 above)
Comment (Rychu-Pawel) [reply cmd/dump.go:58]: And yea, those other places was just me overlooked them. Good catch! :)

**Review by 6543 — APPROVED** [review]

**Review by axifive — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**